x-pack/plugins/fleet/server/services/epm/registry/index.ts

```diff
@@ -256,32 +282,67 @@ function ensureContentType(archivePath: string) {
   return contentType;
 }
 
-export async function ensureCachedArchiveInfo(
-  name: string,
-  version: string,
-  installSource: InstallSource = 'registry'
-) {
-  const paths = getArchiveFilelist({ name, version });
-  if (!paths || paths.length === 0) {
-    if (installSource === 'registry') {
-      await getRegistryPackage(name, version);
-    } else {
-      throw new PackageCacheError(
-        `Package ${name}-${version} not cached. If it was uploaded, try uninstalling and reinstalling manually.`
-      );
+export async function fetchArchiveBuffer({
+  pkgName,
+  pkgVersion,
+  verify,
+  ignoreUnverified = false,
+}: {
+  pkgName: string;
+  pkgVersion: string;
+  verify: boolean;
+  ignoreUnverified?: boolean;
+}): Promise<{
+  archiveBuffer: Buffer;
+  archivePath: string;
+  verificationResult?: PackageVerificationResult;
+}> {
+  const logger = appContextService.getLogger();
+  const { download: archivePath } = await getInfo(pkgName, pkgVersion);
+  const archiveUrl = `${getRegistryUrl()}${archivePath}`;
+  const archiveBuffer = await getResponseStream(archiveUrl).then(streamToBuffer);
+  if (verify) {
+    const verificationResult = await verifyPackageArchiveSignature({
+      pkgName,
+      pkgVersion,
+      pkgArchiveBuffer: archiveBuffer,
+      logger,
+    });
+
+    if (verificationResult.verificationStatus === 'unverified' && !ignoreUnverified) {
```

Comment:
Should the `unverified` value come from an enum?

Reply:
Personally as typescript validates the enum values, I find using plain strings neater in this situation, but I can see the appeals of both ways.